puppet/modules/vault_server/templates/vault.service.erb

```diff
@@ -5,6 +5,7 @@ After=consul.service vault-assets.service
 
 [Service]
 Environment=AWS_REGION=<%= @region %>
+ExecStartPre=/bin/sh -c "while true; do STATUS=$(curl -s -o /dev/null -k -w '%{http_code}' --request GET http://localhost:8500/v1/health/node/<%= @hostname %>); if [ $STATUS -eq 200 ]; then break; else sleep 5; fi; done"
```

Comment:
```
// check for acl readiness
 req, err = http.NewRequest("GET", fmt.Sprintf("%s/v1/acl/token/self", c.HTTPHostAddr), nil)
 if err != nil {
        return fmt.Errorf("error building acl readiness request: %v", err)
}
```

Node healthy does not mean the ACL system is ready. Can you please test it like that?

Also the whole hostname thing is quite ugly